### PR TITLE
Add support for asnprintf to the MemoryLeakDetector

### PR DESCRIPTION
**Pull request by AlanRosenthal**
see: https://man7.org/linux/man-pages/man3/asprintf.3.html

### CODE REVIEW

File: src/CppUTest/TestHarness_c.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 
 extern "C"
 {
+#include <stdio.h>
```
Comment (thetic) [inline src/CppUTest/TestHarness_c.cpp:36]: Direct reliance on the C standard library is not allowed in CppUTest.

**Review by thetic — CHANGES_REQUESTED** [review]
Clang-tidy warnings need to be addressed.
Build without C standard library needs to be repaired.
We need to get test coverage for this before we can consider merging.

File: src/CppUTest/TestHarness_c.cpp (hunk 1 above)
Comment (thetic) [reply src/CppUTest/TestHarness_c.cpp:36]: My intuition is that this feature does not belong in CppUTest, but rather in your own force include. I don't see anything in the implementation that requires it to be tightly integrated with CppUTest itself.

`asnprintf` is non-standard; its a GCC/Linux extension. CppUTest is focused on portability across old, embedded, and resource-constrained systems. `asnprintf` isn't portable and I don't see it as especially valuable for CppUTest's target use-cases.

That's just my opinion though.@basvodde, did you have thoughts?

**Comment by basvodde** [comment]
@AlanRosenthal Why close this?

**Pull request closed without merging**